services/venice-controller/src/main/java/com/linkedin/venice/controller/UnusedValueSchemaCleanupService.java

```diff
@@ -0,0 +1,127 @@
+package com.linkedin.venice.controller;
+
+import com.linkedin.venice.common.VeniceSystemStoreUtils;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.schema.SchemaEntry;
+import com.linkedin.venice.service.AbstractVeniceService;
+import com.linkedin.venice.utils.DaemonThreadFactory;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+
+public class UnusedValueSchemaCleanupService extends AbstractVeniceService {
+  private final ScheduledExecutorService executor =
+      Executors.newSingleThreadScheduledExecutor(new DaemonThreadFactory("UnusedValueSchemaCleanupService"));
+  private final VeniceControllerMultiClusterConfig multiClusterConfig;
+  private final VeniceParentHelixAdmin veniceParentHelixAdmin;
+  private final VeniceHelixAdmin admin;
+  private final int scheduleIntervalMinutes;
+  private boolean stop = false;
+  public final int minSchemaCountToKeep;
+
+  UnusedValueSchemaCleanupService(
+      VeniceControllerMultiClusterConfig multiClusterConfig,
+      VeniceHelixAdmin admin,
+      VeniceParentHelixAdmin parentHelixAdmin) {
+    this.multiClusterConfig = multiClusterConfig;
+    this.admin = admin;
+    this.scheduleIntervalMinutes = multiClusterConfig.getUnusedSchemaCleanupIntervalMinutes();
+    this.minSchemaCountToKeep = multiClusterConfig.getMinSchemaCountToKeep();
+    this.veniceParentHelixAdmin = parentHelixAdmin;
+  }
+
+  private Runnable getRunnableForSchemaCleanup() {
```

Comment:
We should catch and report/log exceptions that may be thrown from schema deletion so we know if the feature is working for all stores or not.

Reply:
Sure